lib/KBaseReport/utils/validation_utils.py

```diff
@@ -67,11 +70,21 @@ def validate_extended_report_params(params):
         'report_object_name': {'type': 'string', 'nullable': True},
         'html_window_height': {'type': 'integer', 'min': 1, 'nullable': True},
         'summary_window_height': {'type': 'integer', 'min': 1, 'nullable': True},
-        'direct_html_link_index': {'type': 'integer', 'min': 0, 'nullable': True},
-        'direct_html': {'type': 'string', 'nullable': True}
+        'direct_html_link_index': {
+            'type': 'integer',
+            'min': 0,
+            'nullable': True,
+            'dependencies': 'html_links',
+        },
```

Comment:
Added checks that `html_links` and `direct_html_link_index` are both present. Not sure this should be `nullable` - might be easier if the validator picked up that issue.

Reply:
Not saying we should switch, but in more recent days I've started to prefer jsonschema. It can handle this kind of stuff well.